components/raytac/MDBT50Q.stanza

```diff
@@ -0,0 +1,211 @@
+    if (div > 0.0) :
+      pad p[i] : smd-pad(padx, pady) at loc(9.9 - offset, 6.0 + (to-double((i - 45) / 2) + 1.0) * pitch)
+    else :
+      pad p[i] : smd-pad(padx, pady) at loc(9.9, 6.0 + (to-double((i - 46) / 2 ) + 1.5) * pitch)
+
+  for (i in 50 to 55) do :
+    val div = to-double(i % 2)
+    if (div < 1.0) :
+      pad p[i] : smd-pad(padx, pady) at loc(9.9 - offset, 9.2 + to-double((i - 50) / 2) * pitch)
+    else :
+      pad p[i] : smd-pad(padx, pady) at loc(9.9, 9.2 + (to-double((i - 51) / 2) + 0.5) * pitch)
+
+  pad p[55] : smd-pad(padx, pady) at loc(9.9, 11.5)
+
+  for (i in 56 to 62) do :
+    pad p[i] : smd-pad(pady, padx) at loc(3.25 + to-double(i - 56) * pitch, 7.2)
+
+  layer(Courtyard(Top)) = Rectangle(10.5, 15.5, loc(10.5 / 2.0, 15.5 / 2.0))
+  layer(ForbidCopper(LayerIndex(0), LayerIndex(0, Bottom))) = Rectangle(12.4, 3.8, loc(10.5 / 2.0, 15.5 - 3.8 / 2.0))
+  layer(ForbidCopper(LayerIndex(0), LayerIndex(0))) = Rectangle(1.6, 1.2, loc(2.95 + 1.6 / 2.0, 15.5 - 3.8 - 1.2 / 2.0))
+  ref-label()
+
+public pcb-module module (antenna:String) :
+  pin gnd
+  pin vcc
+  pin vbus
+  pin reset
+  pin swdio
+  pin swdck
+
+  port spi-master : spi
+  port i2c-master : i2c
+  port uart-nrf : uart
+  port P0 : pin[32]
+  port P1 : pin[10]
+
+  inst ble-module : ocdb/raytac/MDBT50Q/component(antenna)
+
+  ; Power connections and decoupling caps
+  net (gnd, ble-module.GND)
+  net (vcc, ble-module.VDD)
+  net (vcc, ble-module.VDDH)
+  net (vbus, ble-module.VBUS)
+  cap-strap(vcc, gnd, 10.0e-6)
+  cap-strap(vbus, gnd, 10.0e-6)
+
+  ; swd
+  net (swdio, ble-module.SWDIO)
+  net (swdck, ble-module.SWDCK)
+
+  ; spi
+  net (spi-master.mosi, ble-module.P0_28)
+  net (spi-master.miso, ble-module.P0_30)
+  net (spi-master.sck, ble-module.P0_31)
+  net (spi-master.ss, ble-module.P0_02)
+
+  ; i2c
+  net (i2c-master.sda, ble-module.P0_11)
+  net (i2c-master.scl, ble-module.P0_12)
+  res-strap(i2c-master.sda, vcc, 1.2e3)
+  res-strap(i2c-master.scl, vcc, 1.2e3)
+
+  ; uart
+  net (uart-nrf.rx, ble-module.P0_08)
+  net (uart-nrf.tx, ble-module.P0_06)
+  
+  ; 32 kHz crystal
+  inst lf-xtal : ocdb/epson/FC-135/component
+  cap-strap(lf-xtal.p[1], gnd, 9.0e-12)
+  cap-strap(lf-xtal.p[2], gnd, 9.0e-12)
+  net (lf-xtal.p[1], ble-module.P0_00)
+  net (lf-xtal.p[2], ble-module.P0_01)
+
+  ; gpio
+  ; nrf reset
+  net (reset, ble-module.P0_18)
+  res-strap(reset, vcc, 10.0e3)
+
+  val P0-pins = [ble-module.P0_00 ble-module.P0_01 ble-module.P0_02 ble-module.P0_03 ble-module.P0_04 ble-module.P0_05 ble-module.P0_06 ble-module.P0_07 ble-module.P0_08 ble-module.P0_09 ble-module.P0_10 ble-module.P0_11 ble-module.P0_12 ble-module.P0_13 ble-module.P0_14 ble-module.P0_15 ble-module.P0_16 ble-module.P0_17 ble-module.P0_18 ble-module.P0_19 ble-module.P0_20 ble-module.P0_21 ble-module.P0_22 ble-module.P0_23 ble-module.P0_24 ble-module.P0_25 ble-module.P0_26 ble-module.P0_27 ble-module.P0_28 ble-module.P0_29 ble-module.P0_30 ble-module.P0_31]
```

Comment:
If the component had the P0 pins listed as `P0[0]`, `P0[1]`, then they wouldn't have to do this.